### ISSUE TITLE
dofile() won't execute files in Flash when LFS _init.lua is set

### ISSUE THREAD

**Issue by chathurangawijetunge** [nodemcu/nodemcu-firmware#3278]
I have `_init.lua` in my `LFS` if i `dofile("abc.lua")` where `abc.lua` is in my flash not in `LFS` I get a error
``attempt to call a nil value in function 'dofile' ``
this use to work earlier......

**Comment by vsky279** [comment]
I can reproduce the error. It seem that the return parameters of  `node.flashindex` have changed (no second variable is returned).

I think the simplest implementation of both function is:
```Lua
local lf, df = loadfile, dofile
G.loadfile = function(n)
  local mod, ext = n:match("(.*)%.(l[uc]a?)");
  local fn = index(mod)
  if fn then return fn else return lf(n) end
end

G.dofile = function(n)
  local mod, ext = n:match("(.*)%.(l[uc]a?)");
  local fn = index(mod)
  if fn then return fn() else return df(n) end
end
```
We don't need to check the `ext` value for 'lua|lc' instead we can call `df(n)` directly and let the user face directly the error.

@TerryE Terry, could you please have look?

**Comment by bazooka07** [comment]
I have had the same issue. 

I adjust _init.lua as follow :
```lua
package.loaders[3] = function(module) -- loader_flash
  return lfs_t[module]
end

local lf, df = loadfile, dofile
G.loadfile = function(n)
  local mod, ext = n:match("(.*)%.(l[uc]a?)");
  if ext ~= 'lc' and ext ~= 'lua' then return lf(n) end
  local fn   = index(mod)
  if fn then return fn else return lf(n) end
end

G.dofile = function(n)
  local mod, ext = n:match("(.*)%.(l[uc]a?)");
  if ext ~= 'lc' and ext ~= 'lua' then return df(n) end
  local fn = index(mod)
  if fn then return fn else return df(n) end
end
```
Notice that "local index = node.flashindex" is deprecated with the last release, but _init.lua is not updated. So, replace as follows :
```lua
local index = node.LFS.get
```
I'm using LUA 5.1. Not testing with LUA 5.3

**Comment by nwf** [comment]
I believe this was just fixed in #3290 .  Please scream if that isn't true.
